src/Sound/Tidal/Stream.hs

```diff
@@ -213,28 +212,27 @@ startStream :: Config -> [(Target, [OSC])] -> IO Stream
 startStream config oscmap
   = do sMapMV <- newMVar Map.empty
        pMapMV <- newMVar Map.empty
-       bussesMV <- newMVar []
        globalFMV <- newMVar id
        actionsMV <- newEmptyMVar
 
        tidal_status_string >>= verbose config
        verbose config $ "Listening for external controls on " ++ cCtrlAddr config ++ ":" ++ show (cCtrlPort config)
        listen <- openListener config
 
-       cxs <- mapM (\(target, os) -> do remote_addr <- resolve (oAddress target) (show $ oPort target)
-                                        remote_bus_addr <- if isJust $ oBusPort target
-                                                           then Just <$> resolve (oAddress target) (show $ fromJust $ oBusPort target)
-                                                           else return Nothing
+       cxs <- mapM (\(target, os) -> do remote_addr <- resolve (oAddress target) (oPort target)
+                                        remote_bus_addr <- sequence (resolve (oAddress target) <$> (oBusPort target))
```

Comment:
Smart! This uses the fact that Maybe is a Monad so sequence is the same thing as what was done here before.